cache/class-vary-cache.php

```diff
@@ -414,24 +416,28 @@ private static function parse_nocache_cookie() {
 	 */
 	private static function parse_group_cookie() {
 		// If the cache layer supplies a decrypted segmentation header, use that instead of decrypting it again.
-		if ( self::is_encryption_enabled() && isset( $_SERVER[ self::HEADER_AUTH ] ) &&  ! empty( $_SERVER[ self::HEADER_AUTH ] ) ) {
+		if ( self::is_encryption_enabled() && ! empty( $_SERVER[ self::HEADER_AUTH ] ) ) {
+			// phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
 			$cookie_value = $_SERVER[ self::HEADER_AUTH ];
-		} elseif ( self::is_encryption_enabled() && ! empty( $_COOKIE[ self::COOKIE_AUTH ] ) ) {
+		} elseif ( self::is_encryption_enabled() && ! empty( $_COOKIE[ self::COOKIE_AUTH ] ) && isset( $_SERVER['HTTP_COOKIE'] ) ) {
```

Comment:
We didn't check if `$_SERVER['HTTP_COOKIE']` is set; the lack of the check can cause warnings in WP CLI.